### PR TITLE
refactor: simplify authz

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
- Make DefaultAuthorizationRequirement a superclass to simplify calls in inherited requirements
- Simplify some handlers and requirements

**Issues**

### CODE REVIEW

File: Jellyfin.Api/Auth/DefaultAuthorizationPolicy/DefaultAuthorizationRequirement.cs [hunk 1]
```diff
@@ -7,5 +7,18 @@ namespace Jellyfin.Api.Auth.DefaultAuthorizationPolicy
...
+        /// <param name="validateParentalSchedule">A value indicating whether to validate parental schedule.</param>
+        public DefaultAuthorizationRequirement(bool validateParentalSchedule = true)
+        {
+            ValidateParentalSchedule = validateParentalSchedule;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether to ignore parental schedule.
+        /// </summary>
+        public bool ValidateParentalSchedule { get; init; }
```
Comment (barronpm) [inline Jellyfin.Api/Auth/DefaultAuthorizationPolicy/DefaultAuthorizationRequirement.cs:22]: If you're setting this in the constructor, is there any reason to also include the `init`?

Comment (cvium) [reply Jellyfin.Api/Auth/DefaultAuthorizationPolicy/DefaultAuthorizationRequirement.cs:22]: No, it was from a previous iteration

File: Jellyfin.Data/DayOfWeekHelper.cs [hunk 2]
```diff
@@ -17,5 +17,16 @@ public static DayOfWeek[] GetDaysOfWeek(DynamicDayOfWeek day)
                 _ => new[] { (DayOfWeek)day }
             };
         }
+
+        public static bool Contains(this DynamicDayOfWeek dynamicDayOfWeek, DayOfWeek dayOfWeek)
+        {
+            return dynamicDayOfWeek switch
+            {
+                DynamicDayOfWeek.Everyday => true,
+                DynamicDayOfWeek.Weekday => dayOfWeek is > DayOfWeek.Sunday and <= DayOfWeek.Friday,
```
Comment (cvium) [inline Jellyfin.Data/DayOfWeekHelper.cs:26]:
```suggestion
                DynamicDayOfWeek.Weekday => dayOfWeek is >= DayOfWeek.Monday and <= DayOfWeek.Friday,
```

**Review by Shadowghost — COMMENTED** [review]
Functionality wise I verified this use cases:
* Parental control (working)
* Downloading (working)
* Elevation (working)

One thing I noticed though: When reloading the page I always need to log in again.

**Review by nielsvanvelzen — DISMISSED** [review]
The OpenAPI specification is missing a lot (but not all) 401/403 responses with this PR. They need to be added back.

**Review by crobibero — APPROVED** [review]

**Review by Shadowghost — CHANGES_REQUESTED** [review]
When reloading the page as a non-admin user I always need to log in again.

**Review by Shadowghost — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

**Pull request merged**